keepmenu

```diff
@@ -241,6 +244,56 @@ def dmenu_err(prompt):
     return dmenu_select(1, prompt)
 
 
+
+def get_passwd_settings():
+    """
+    Read settings for password generation
+
+    Returns: (dict of presets)
+    """
+    presets = {}
+    if CONF.has_section('password'):
+        args = CONF.items('password')
+        args_dict = dict(args)
+
+        for key, value in args_dict.items():
+            groups = value.split('+')
+            passwd_set = set()
+            for group in groups:
+                group = group.strip()
+                if group in CHAR_TYPES:
+                    passwd_set.add(group)
+                else:
+                    dmenu_err(f"Error in password preset {key}")
+                presets[key] = passwd_set
+    if not presets:
+        presets["letters+digits+punctuation"] = ["letters","digits","braces","punctuation","quotes","dashes","mathsym","logograms"]
+        presets["letters+digits"] = ["letters","digits"]
+        presets["letters"] = ["letters"]
+    return presets
+
+CHAR_TYPES = {
+    "letters" : string.ascii_letters,
+    "digits" : string.digits,
+    "braces" : '(){}[]',
+    "punctuation" : '.,:;',
+    "quotes" : '"\'',
+    "dashes" : '\\-|_/',
+    "mathsym" : '<*+!?=',
+    "logograms" : '#$%&@^`~',
```

Comment:
Please run the code through pylint and flake8 for formatting/style corrections. Although I do use a max line length of 100 instead of 80. Thanks!

Reply:
Totally forgot about this! Though it produces like 50+ warnings from pylint and flake8. I hopefully can fix all of those but it would kinda bloat this PR. What do you think?